plugins/inventory/azure_rm.py

```diff
@@ -444,6 +469,7 @@ def _on_arcvm_page_response(self, response):
             url = '{0}/providers/Microsoft.AzureStackHCI/virtualMachineInstances'.format(arcvm['id'])
             # Stack HCI instances look close enough to regular VMs that we can share the handler impl...
             self._enqueue_get(url=url, api_version=self._stackhci_api_version, handler=self._on_vm_page_response, handler_args=dict(arcvm=arcvm))
+            self._hosts.append(ArcHost(arcvm, self, legacy_name=self._legacy_hostnames))
```

Comment:
i think you need to remove this line?
otherwise i'm getting: 
> Could not set ansible_host for host S00004BK1: No first item, sequence was empty.

[due to my hostvar_expressions referencing (private_ipv4_addresses | first) ]

but i do agree this is working fine:
`ansible-inventory -i inventories/HCI/ --list --limit S00004BK1` if i remove my hostvar_expressions 


however you completely splitted classic arc vm & stack hci vm (which makes sense) thus i think it's better to remove this line altogether

Reply:
Note: yeah if you fill private_ipv4_addresses you can keep this line, the hostvar_expressions works

Reply:
I'm going to remove this since I think it should be requested via include_arc_resource_groups